### PR TITLE
Add detection for zsh completion files

### PR DESCRIPTION
**Pull request by griano**
## Chenges

* In ZSH completions files start with `#compdef` and `#autoload`, like [these files](https://github.com/zsh-users/zsh-completions/tree/master/src)
* I also added detection for some typical ZSH env files.
## Testing

* `rake` passes all tests
* Example with ``#autoload``
```bash
$ ./bin/rougify guess =(curl -sq https://github.com/zsh-users/zsh/blob/master/Completion/Base/Utility/_arguments)
{ tag: "html", title: "HTML", desc: "HTML, the markup language of the web" }
``` 
* Example with ``#compdef``
```bash
$ ./bin/rougify guess =(curl -sq https://github.com/zsh-users/zsh/blob/master/Completion/Zsh/Command/_alias)                                                                               
{ tag: "html", title: "HTML", desc: "HTML, the markup language of the web" }
```

### CODE REVIEW

File: lib/rouge/lexers/shell.rb [hunk 1]
```diff
@@ -9,15 +9,17 @@ class Shell < RegexLexer
 
       tag 'shell'
       aliases 'bash', 'zsh', 'ksh', 'sh'
-      filenames '*.sh', '*.bash', '*.zsh', '*.ksh', '.bashrc', '.zshrc',
-                '.kshrc', '.profile', 'APKBUILD', 'PKGBUILD', '*.ebuild',
+      filenames '*.sh', '*.bash', '*.zsh', '*.ksh', '.bashrc', '.zshrc', 'zshrc',
+                '.kshrc', '.profile', '.zshenv', 'zshenv', '.zprofile', 'zprofile',
+                'APKBUILD', 'PKGBUILD', '*.ebuild',
```
Comment (tancnle) [inline lib/rouge/lexers/shell.rb:14]: **note:** Additional zsh files can be found at https://zsh.sourceforge.io/Doc/Release/Files.html#Files-1

**Review by tancnle — COMMENTED** [review]
Thank you for your contribution @griano ❤️ I have a small suggestion for your consideration. 

Additionally, would it be possible if you can add some tests for the new guessing logic? You can find some existing examples [here](https://github.com/rouge-ruby/rouge/blob/master/spec/lexers/shell_spec.rb#L36). Let me know if you need any assistance.

**Comment by tancnle** [comment]
Thank you for considering the suggestions, @griano. The change LGTM 🚀

**Comment by griano** [comment]
Addressed the requested changes.
### Testing:

The test above were reported incorrectly. In these tests I use the ``raw`` URL to download the actual file.

Test with `#autoload`
```bash
$ bin/rougify guess <(curl -sq https://raw.githubusercontent.com/zsh-users/zsh/master/Completion/Base/Utility/_arguments) 
{ tag: "shell", title: "shell", desc: "Various shell languages, including sh and bash" }
```
Test with `#compdef``
```bash
$ bin/rougify guess <(curl -qs https://raw.githubusercontent.com/zsh-users/zsh/master/Completion/Zsh/Command/_alias)
{ tag: "shell", title: "shell", desc: "Various shell languages, including sh and bash" }
```bash

**Pull request merged**